packages/camera/camera_android_camerax/lib/src/resolution_strategy.dart

```diff
@@ -0,0 +1,181 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show immutable;
+
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// The resolution strategy defines the resolution selection sequence to select
+/// the best size.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/resolutionselector/ResolutionStrategy.
+@immutable
+class ResolutionStrategy extends JavaObject {
+  /// Construct a [ResolutionStrategy].
+  ResolutionStrategy({
```

Comment:
What do you think about making the arguments to this constructor non-null (as they are for the android class) and instead using a private named constructor instead for the creation of the highest available strategy?

Reply:
I love this idea. Changed it and added tests.

Reply:
I believe as is, it is still nullable (it is `required` but someone can still pass null). 

I think this can be changed to `required Size this.boundSize`, though, to keep it as an initializing formal but just with a stricter (non-nullable) type. 

See the "NOTE: Also note that it is possible to enforce a type that is stricter"... in https://dart.dev/tools/linter-rules/prefer_initializing_formals (can't link sections unfortunately).